src/components/ConnectedMap/map-libre-adapter/index.tsx

```diff
@@ -2,9 +2,13 @@ import React, { useEffect, useRef, useState, forwardRef } from 'react';
 import mapLibre from 'maplibre-gl';
 import 'maplibre-gl/dist/maplibre-gl.css';
 import { configRepo } from '~core/config';
-import { currentMapPositionAtom } from '~core/shared_state';
 import { EVENT_MAP_IDLE } from '~core/metrics/constants';
 import { dispatchMetricsEvent } from '~core/metrics/dispatch';
+import {
+  currentMapPositionAtom,
+  setCurrentMapPosition,
+} from '~core/shared_state/currentMapPosition';
+import { store } from '~core/store/store';
```

Comment:
you have useAtom hook, store not needed in react components/hooks

Reply:
Is it worth using `useAtom` hook here, considering its value is only really used on map initialization with `extent`?

Reply:
Added useAction for `setCurrentMapPosition`.

Using useAtom doesn't make sense here, because we only need the value of the atom once, on the startup. Adding this dependency to useEffect() will cause it to fire aimlessly each time we move the camera (just checked it)

Reply:
Fixed. Of course it was enough to just use `ctx.spy`, should have thought if this sooner